### PR TITLE
Pull request for React GUI and main.py refactoring

### PR DESCRIPTION
**Pull request by varkha-d-sharma**
# Related Issues / Pull Requests

List all related issues and/or pull requests if there are any.

# Description

Include a brief summary of the proposed changes.

# What changes are proposed in this pull request?

- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected; for instance, 
      examples in this repository need to be updated too).
- [ ] This change requires a documentation update.

# Checklist:

- [ ] My code follows the style guidelines of this project (PEP-8 with Google-style docstrings).
- [ ] My code modifies existing public API, or introduces new public API, and I updated or wrote docstrings that
      uses Google-style formatting and any other formatting that is supported by mkdocs and plugins this project
      uses.
- [ ] I have commented my code.
- [ ] My code requires documentation updates, and I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.

### CODE REVIEW

File: docs/cmf_server/cmf-server.md [hunk 1]
```diff
@@ -26,10 +26,46 @@ They accept and return JSON-encoded request bodies and responses and return stan
...
+    volumes:
+      - /home/<user>/cmf-server/data:/cmf-server/data
+    container_name: cmf-server
+    build:
+    ....
+  ```
+  
+ 4. Execute following command to start both the containers. `IP` variable is the IP address of the machine on which you are executing the following command.
+    ```
+    IP=200.200.200.200 docker compose -f docker-compose-server.yml up
```
Comment (annmary-roy) [inline docs/cmf_server/cmf-server.md:60]: we need to remove the dependency on the IP. it can be a host name also

File: server/app/get_data.py [hunk 2]
```diff
@@ -14,30 +17,118 @@ def get_executions(mlmdfilepath, pipeline_name):
...
+            executions = query.get_all_executions_in_stage(stage)
+            dict_executions = executions.to_dict("dict")  # converting it to dictionary
+            identifiers.append(dict_executions["id"][0])
     name = []
     url = []
     df = pd.DataFrame()
     for identifier in identifiers:
-        get_artifacts = query.get_all_artifacts_for_execution(identifier)  # getting all artifacts
-        artifacts_dict = get_artifacts.to_dict('dict')  # converting it to dictionary
+        get_artifacts = query.get_all_artifacts_for_execution(
```
Comment (annmary-roy) [inline server/app/get_data.py:34]: we need to run performance measurement to check the most ideal way of querying. Another approach would be to get all artifacts in the store and filter them by pipeline name. we need to measure the performance at scale

File: server/app/get_data.py [hunk 3]
```diff
@@ -14,30 +17,118 @@ def get_executions(mlmdfilepath, pipeline_name):
 
 
 # This function fetches all the artifacts available in given mlmd
-def get_artifacts(mlmdfilepath, pipeline_name,data):
+def get_artifacts(mlmdfilepath, pipeline_name, data):
     query = cmfquery.CmfQuery(mlmdfilepath)
```
Comment (annmary-roy) [inline server/app/get_data.py:21]: what is data parameter in this ? please add a comment for that

File: server/app/get_data.py [hunk 4]
```diff
@@ -14,30 +17,118 @@ def get_executions(mlmdfilepath, pipeline_name):
...
 
+def create_unique_executions(server_store_path, req_info):
+    mlmd_data = json.loads(req_info["json_payload"])
+    pipelines = mlmd_data["Pipeline"]
+    pipeline = pipelines[0]
+    pipeline_name = pipeline["name"]
+    executions_server = []
+    list_executions_exists = []
+    if os.path.exists(server_store_path):
+        query = cmfquery.CmfQuery(server_store_path)
```
Comment (annmary-roy) [inline server/app/get_data.py:56]: we should not do this for every call. This should be cached

**Review by annmary-roy — APPROVED** [review]

File: docs/cmf_server/cmf-server.md (hunk 1 above)
Comment (varkha-d-sharma) [reply docs/cmf_server/cmf-server.md:60]: Will do

File: docs/cmf_server/cmf-server.md [hunk 5]
```diff
@@ -26,10 +26,46 @@ They accept and return JSON-encoded request bodies and responses and return stan
 
 ## Setup a cmf-server
 
+There were are two ways to start cmf server -
```
[the remark this answers is not in the document]
Comment (varkha-d-sharma) [reply docs/cmf_server/cmf-server.md:29]: Done

**Pull request merged**